src/server/login.js

```diff
@@ -53,14 +51,27 @@ module.exports = function (client, server, options) {
 
       try {
         const publicKey = crypto.createPublicKey({ key: packet.signature.publicKey, format: 'der', type: 'spki' })
-        const publicPEM = mcPubKeyToPem(packet.signature.publicKey)
-        const signable = packet.signature.timestamp + publicPEM // (expires at + publicKey)
-
-        if (!crypto.verify('RSA-SHA1', Buffer.from(signable, 'utf8'), mojangPubKey, packet.signature.signature)) {
-          raise('multiplayer.disconnect.invalid_public_key_signature')
-          return
+        const publicDER = publicKey.export({ type: 'spki', format: 'der' })
```

Comment:
Not sure what part of this is 1.19.3 and 1.19.2, but looks like this can be simplified (see how it's handled in the other PR)

Reply:
The other PR's validation does not work as the signature is generated from the public key without padding, but client provides it with padding.